### ISSUE TITLE
Misleading error hint about invalid configuration file keys

### ISSUE THREAD

**Issue by pradyunsg** [astral-sh/ruff#12359]
Version: 0.5.2

#### Reproduction instructions

1. Create a `ruff.toml` file with the contents:

    ```toml
    extend-select = [
        "W", # pycodestyle warnings
        "I", # isort
        "N", # pep8-naming
        "D", # pydocstyle
        "B", # flake8-bugbear
        "PYI" # flake8-pyi
    ]

    # Group errors by file.
    output-format = "grouped"

    # Show source context of error.
    show-source = true

    # Show sumamry of fixes.
    show-fixes = true
    ```

2. Create a `src/foo.py` file with no contents.
3. Run `ruff check` and see the problematic error message.

#### Output

```sh-session
❯ ruff check
ruff failed
  Cause: Failed to parse /Users/[...]/ruff.toml
  Cause: TOML parse error at line 1, column 1
  |
1 | extend-select = [
  | ^^^^^^^^^^^^^^^^^
unknown field `show-source`
```

The error message points to the line about `extend-select` when complaining about `show-source` as the key that is unknown. It would be better for the error to point at line 14 entirely OR, ideally, point only at `show-source` part of that line since that's the key in the key-value pair in the configuration file that is unknown.

**Comment by MichaReiser** [comment]
That makes sense. Although I don't think there's much that we can do about it other than writing our own toml parser. The error message comes directly from the `toml` crate 

https://github.com/astral-sh/ruff/blob/117203f71356718b35579ebc6e9805c291dc50cf/crates/ruff_workspace/src/pyproject.rs#L52-L55

**Comment by MichaReiser** [comment]
Could be related to https://github.com/toml-rs/toml/issues/589

**Comment by charliermarsh** [comment]
Yeah that's a shame. Thanks @pradyunsg.

**Comment by chirizxc** [comment]
Here's the original problem https://github.com/serde-rs/serde/issues/1183
